ds3-sdk/src/main/java/com/spectralogic/ds3client/helpers/Ds3ClientHelpersImpl.java

```diff
@@ -246,6 +248,20 @@ public Ds3ClientHelpersImpl(final Ds3Client client,
                 this.eventRunner);
     }
 
+    /**
+     * Verifies that the specified job is active. If the job is not active, then a
+     * JobRecoveryNotActiveException is thrown.
+     * @throws IOException
+     * @throws JobRecoveryNotActiveException
+     */
+    private void innerVerifyJobActive(final UUID jobId) throws IOException, JobRecoveryNotActiveException {
+        try {
+            client.getActiveJobSpectraS3(new GetActiveJobSpectraS3Request(jobId));
+        } catch (final FailedRequestException e) {
```

Comment:
I would only create a different exception if the request returns a 404, otherwise, I would not catch this exception.